### ISSUE TITLE
429 After first request?

### ISSUE THREAD

**Issue by alexsullivan114** [GeneralMills/pytrends#243]
Howdy,

I'm attempting to use the library and I'm getting hit with a 429 error after copying the example code.
Here's my script:

```
from pytrends.request import TrendReq

pytrends = TrendReq(hl='en-US', tz=360)
kw_list = ["Blockchain"]
pytrends.build_payload(kw_list, cat=0, timeframe='today 5-y', geo='', gprop='')
pytrends.interest_over_time()
```

I can visit the trends website fine, and I can copy and paste the URL produced by the API and get a json file just fine. It's hard for me to imagine being rate limited on my first request and still being able to visit the site normally. 

Any ideas?

**Comment by jersobh** [comment]
same problem here

**Comment by alexsullivan114** [comment]
For what its worth, this is _also_ happening to me for an analogous javascript google trends library.

**Comment by hkpoint** [comment]
Same problem here. Seems it only started happen today...

**Comment by alexsullivan114** [comment]
After doing a bit more sleuthing, this is _also_ happening on the last working version I had used, which was version `4.1.1` of the library. Sounds like Google made some changes today.

**Comment by tdrobbin** [comment]
also getting the same problem today

**Comment by ratcliffej** [comment]
Same problem here. :(

**Comment by lopezdp** [comment]
I have been fighting with this all day. Even tried hitting it as such:

`import urllib.request 
with urllib.request.urlopen('https://trends.google.com/trends/explore?date=2009-01-03%202018-05-28&q=bitcoin') as response:
    trnData = response.read()`

I get the same 429...

This hurts, Totally killed progress on my research today.

**Comment by peeyush-ad2games** [comment]
Did anyone manage to find any solution?

**Comment by RKushnir** [comment]
The `/trends/explore` page also fails if you try to open it in a real browser in the incognito mode.
That is because google trends expect some cookie to be present. And if you refresh the page, the cookie is now present and the page works.

So, in the code you would need to simulate this behavior — make the initial request, remember the cookie, then make the same request again, but this time providing the stored cookie.

**Comment by dabdada** [comment]
@RKushnir is this possible as of now? I do not see in the code how.

I tried by changing the code and call trends.google.com in the init of TrendReq call and store it in a cookies attribute. In the get_data I did then provide cookies to requests.get call.

This works, but I am not sure if this is needed. Neither do I know if this should even be handled by TrendReq itself (additional call to google on all TrendReq inits), or shall be provided by the user somehow (for the user it seems kinda inconvinient to think about that every time he requests something)

**Comment by RKushnir** [comment]
@dabdada I mean it as a hint how to fix/work around this issue in general. Probably, the library code needs a fix.

**Comment by RKushnir** [comment]
@zachokeeffe In my experiments the name of the cookie was `NID`, but I was getting it not from `apis.google.com`, but from `trends.google.com`.

**Comment by kakarla7** [comment]
I face the same issue now. If it's worth, I tried to clone the zip file and use it from there it seems to work. 
Not entirely sure what's the reason, though!
![image](https://user-images.githubusercontent.com/17584245/41056078-095cb71c-6991-11e8-8291-0faa66f76f53.png)

**Comment by jersobh** [comment]
@kakarla7 so do a # sudo pip install pytrends --upgrade 
(to install it globally or just do it on your venv. it may be pip3 or pip3.6 also). The latest version seems to work fine.

**Comment by dripti** [comment]
Anyone with any luck on it? I am getting the same issue

**Comment by ratcliffej** [comment]
Make sure you download the latest version of Pytrends from this repo. I had this problem, and the last check-in seems to have fixed it.

**Comment by kakarla7** [comment]
@dripti yes. directly installing from pytrends githu repo helped me (picture in my previous comment)

**Comment by dabdada** [comment]
Could you maybe try deleting pyc files?

**Comment by polzkey** [comment]
I am experiencing the same problem. I tried removing and installing it again but did not fix the problem. Tried updating and installing directly from the repo did not fix the problem.

**Comment by lopezdp** [comment]
Here is how I solved it:

https://stackoverflow.com/questions/50571317/pytrends-the-request-failed-google-returned-a-response-with-code-429/50616233#50616233

**Comment by MYRepoMan** [comment]
Anyone know how long you go to rate limit jail? I can't even do a google trends web search without error 429.

**Comment by RKushnir** [comment]
@MYRepoMan do you have cookies enabled? Maybe some browser add-on is blocking them?

**Comment by rsmahabir** [comment]
I installed the package from pycharm and it works for me

**Comment by rsmahabir** [comment]
May be easier to clone the repository and install using the setup.py file

**Comment by rsmahabir** [comment]
Its looks like request.py was updated. I switched out that file and everything seems to work without issue :)

**Comment by kalagee-anjaria** [comment]
Hello all,
I am still getting the same error even though my pytrends library is of the latest version. Any suggestions on what should I do now to get rid of this error?

**Comment by dabdada** [comment]
can you provide an error message for what you are seeing and some code @kalagee-anjaria

**Comment by kalagee-anjaria** [comment]
The error message is The request failed: Google returned a response with code 429. The pytrends = TrendReq works fine in my localhost, but on AWS server it keeps giving this error.

**Comment by dabdada** [comment]
what aws setup do you use? lambda, ec2, anything else?

**Comment by kalagee-anjaria** [comment]
I am using EC2

**Comment by JessHPRH** [comment]
I've also found the problem again in my localhost. I'm using jupyter notebook. 

I get this:
`ResponseError: The request failed: Google returned a response with code 429.`

I've tried making sure it's the most up-to-date version of `pytrends` as per [this comment](https://stackoverflow.com/a/50616233/11937086).

**Comment by kalagee-anjaria** [comment]
@JessHPRH  Try using VS Code. It's working on localhost there

**Comment by JessHPRH** [comment]
I'm using Anaconda and when I tried to run my script in VS Code it told me it didn't recognise my installation of `numpy`. I don't fancy reconfiguring my whole system to make one script work :/

**Comment by eracle** [comment]
The problem  comes from Google bot recognition system thatwhich tries to throttle down requests that do not have convincing cookies.

I solved in the following way:
1. First of all you must use google trends from your web browser, then track the actual HTTP get made;
2. Right-click the request inside the network tab (I am using Chromium, and the request is named: /trends/explore?q=topic&geo=US ) and select Copy->Copy as cURL;
3. Then go to https://curl.trillworks.com/ and paste the curl script on the left side of the page e copy the python script generated on the right side of the page;
4. Then subclass the TrendReq class in order to pass the custom header just copied:

````python
from pytrends.request import TrendReq as UTrendReq
GET_METHOD='get'

import requests

headers = {
...
}


class TrendReq(UTrendReq):
    def _get_data(self, url, method=GET_METHOD, trim_chars=0, **kwargs):
        return super()._get_data(url, method=GET_METHOD, trim_chars=trim_chars, headers=headers, **kwargs)

````

5. Then retry again removing any import TrendReq from your code but instead use this just created.

**Comment by eracle** [comment]
TLDR; I solved the problem with a custom patch
You can find the patch on [this](https://stackoverflow.com/a/67199394/3289963) link.
Upvotes are strongly appreciated.

**Comment by siddhart1o1** [comment]
> what aws setup do you use? lambda, ec2, anything else?

I want to use pytrend with lamda but getting error.

**Comment by Lockley139** [comment]
> TLDR; I solved the problem with a custom patch You can find the patch on [this](https://stackoverflow.com/a/67199394/3289963) link. Upvotes are strongly appreciated.

This didnt work for me unfortunately, after implementing into my script :(
